pkg/apicast/apicast_option_provider.go

```diff
@@ -122,6 +122,26 @@ func (a *APIcastOptionsProvider) GetApicastOptions() (*APIcastOptions, error) {
 	a.APIcastOptions.Workers = a.APIcastCR.Spec.Workers
 	a.APIcastOptions.Timezone = a.APIcastCR.Spec.Timezone
 
+	for idx, customPolicySpec := range a.APIcastCR.Spec.CustomPolicies {
+		namespacedName := types.NamespacedName{
+			Name:      customPolicySpec.SecretRef.Name, // CR Validation ensures not nil
```

Comment:
Even though CR validation ensures that `customPolicySpec.SecretRef` is not nil, unfortunately it does not ensure that the `Name` is not empty. We should check that name is not empty too. Maybe inside validateCustomPolicySecret method

Reply:
validated